ios/RNHeap.m

```diff
@@ -14,7 +22,30 @@ @implementation RNHeap
 static BOOL appIdSet = NO;
 
 - (dispatch_queue_t)methodQueue {
-  return dispatch_get_main_queue();
+    return dispatch_get_main_queue();
+}
+
+static void(^delayedPageview)(void);
+
+/// This function delays a pageview until we think the animation has completed.
+/// Since the state change fires at the start of the animation, we wait half a second for the transiton to complete.
++ (void)scheduleDelayedPageview:(void(^)(void))block
+{
+    [self consumeDelayedPageview];
```

Comment:
I'm not sure I completely understand this line. Is this to flush out a prior pageview?

Reply:
I went ahead and simplified things so there's no need for this.  Previously, I was thinking I could consume pageviews early when `viewDidAppear:` was called, but it just ends up firing early so there was no use for the complexity.